c/include/libsbp/cpp/state.h

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (C) 2019 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#ifndef SBP_CPP_STATE_H
+#define SBP_CPP_STATE_H
+
+#include <libsbp/sbp.h>
+
+namespace sbp {
+
+class IReader {
+ public:
+  virtual ~IReader() = default;
+
+  virtual s32 read(u8 *buffer, u32 buffer_length) = 0;
+};
+
+class IWriter {
+ public:
+  virtual ~IWriter() = default;
+
+  virtual s32 write(const u8 *buffer, u32 buffer_length) = 0;
+};
+
+class State {
```

Comment:
The name came from the `sbp_state_t` which it wraps into a C++ object. I'm open to alternate suggestions though.

Reply:
I would actually call it `SBP` since this class is how I think of SBP but it might be inappropriate. I'll leave it up to you, my comment was merely a suggestion